app/reflexes/admin/orders_reflex.rb

```diff
@@ -27,13 +27,16 @@ def ship
     end
 
     def bulk_invoice(params)
+      visible_orders = editable_orders.where(id: params[:bulk_ids]).filter(&:invoiceable?)
+      return unless all_distributors_can_invoice?(visible_orders)
+
       cable_ready.append(
         selector: "#orders-index",
         html: render(partial: "spree/admin/orders/bulk/invoice_modal")
       ).broadcast
 
       BulkInvoiceJob.perform_later(
-        params[:bulk_ids],
```

Comment:
we're printing any invoice requested by the users without any validation.
This opens access to all the invoices including those that belong to the other users.

(BulkInvoiceJob doesn't do any validation, it just sorts and prints)

Reply:
That's fixed by the use of `editable_orders` above, right ?

Reply:
yes, correct.